test/tests/functional/pbs_dup_acc_log_for_resv.py

```diff
@@ -52,50 +52,45 @@ def setUp(self):
             self.skipTest('test requires two MoMs as input, ' +
                           'use -p moms=<mom1>:<mom2>')
 
-    def check_log_resv_time(self, accounting_log_msg, start_tup,
-                            end_tup, check_status):
+    def check_log_resv_time(self, accounting_log_msg, start_tup):
         """
-        This is a helper function to reterive start and end time
-        of standing rervation from accounting log.
+        This is a helper function to retrieve start and end time
+        of standing reservation from accounting log.
         """
-        loglist_value = accounting_log_msg[start_tup][end_tup]
+        loglist_value = accounting_log_msg[start_tup][1]
         log_value = loglist_value.split()
-        if check_status == "start":
+        actual_reservation_time = []
+        check_status = ['start', 'end']
+        for status in check_status:
             for attribute in log_value:
                 attribute_value = attribute.split("=")
-                if attribute_value[0] == 'start':
-                    actual_reservation_time = attribute_value[1]
-        elif check_status == "end":
-            for attribute in log_value:
-                attribute_value = attribute.split("=")
-                if attribute_value[0] == 'end':
-                    actual_reservation_time = attribute_value[1]
-        return int(actual_reservation_time)
+                if attribute_value[0] == status:
+                    actual_reservation_time.append(int(attribute_value[1]))
```

Comment:
1) this for loop is not efficient. It will continue to iterate even after status is found in log_value.
2) We can find both "start" and "end" in one pass itself. You are making two passes through the log_value which can be avoided.